torchx/lib/torchx/backend.ex

```diff
@@ -195,14 +232,25 @@ defmodule Torchx.Backend do
   end
 
   @impl true
-  def slice(%T{shape: shape} = out, %T{} = t, start_indices, lengths, strides) do
+  def slice(
+        %T{shape: output_shape} = out,
+        %T{shape: input_shape} = t,
+        start_indices,
+        lengths,
+        strides
+      ) do
     t
     |> from_nx()
-    |> narrow(start_indices, lengths, 0, shape)
-    |> stride(shape, lengths, strides)
+    |> do_slice(input_shape, output_shape, start_indices, lengths, strides)
     |> to_nx(out)
   end
 
+  defp do_slice(t, input_shape, output_shape, start_indices, lengths, strides) do
+    t
+    |> narrow(start_indices, lengths, 0, input_shape)
+    |> stride(output_shape, lengths, strides)
+  end
```

Comment:
I recommend avoiding the `do_` whenever possible cause it is not very clear. My suggestion:

```suggestion
  defp torchx_slice(ref, input_shape, output_shape, start_indices, lengths, strides) do
    ref
    |> narrow(start_indices, lengths, 0, input_shape)
    |> stride(output_shape, lengths, strides)
  end
```